src/main/java/com/es/phoneshop/model/filter/Filter.java

```diff
@@ -0,0 +1,100 @@
+package com.es.phoneshop.model.filter;
```

Comment:
Move Filter, SortingComparator , FilterCreator, FilterMatcher from model package.
Create comparator package on the root level and add there you comparator implementations. You can create filter package in comparator package and move filter related classes there